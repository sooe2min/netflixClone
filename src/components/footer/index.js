import React from 'react'
import { Container, Title, Row, Column, Link, Select, Text, Copy, Break } from './styles/footer'

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Footer.Title = function FooterTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Footer.Row = function FooterRow({ children, ...restProps }) {
  return <Row {...restProps}>{children}</Row>
}

Footer.Column = function FooterColumn({ children, ...restProps }) {
  return <Column {...restProps}>{children}</Column>
}

Footer.Link = function FooterLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Footer.Select = function FooterSelect({ children, ...restProps }) {
  return <Select {...restProps}>{children}</Select>
}

Footer.Text = function FooterText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Footer.Copy = function FooterCopy({ children, ...restProps }) {
  return <Copy {...restProps}>{children}</Copy>
}

Footer.Break = function FooterBreak({ children, ...restProps }) {
  return <Break {...restProps}>{children}</Break>
}
