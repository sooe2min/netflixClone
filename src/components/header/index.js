import React, { useState } from 'react'
import { Background, Container, Logo, ButtonLink, Group, Link, Profile, Search, SearchIcon, SearchInput, Picture, Dropdown, Feature, FeatureCallOut, Text, PlayButton } from './styles/header'
import { Link as ReactRouterLink } from 'react-router-dom'

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>
}

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>
}

Header.Search = function HeaderSearch({ searchTurm, setSearchTurm, ...restProps }) {
  const [searchActive, setSearchActive] = useState(false)

  return (
    <Search {...restProps}>
      <SearchIcon onClick={() => setSearchActive(!searchActive)}>
        <img src='/images/icons/search.png' alt='search' />
      </SearchIcon>
      <SearchInput
        placeholder='Search films and series'
        value={searchTurm}
        onChange={({ target }) => setSearchTurm(target.value)}
        active={searchActive}
      />
    </Search >
  )
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>
}


Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>
}
