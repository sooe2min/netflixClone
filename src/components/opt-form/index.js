import React from 'react'
import { Container, Text, Input, Button, Break } from './styles/opt-form'

export default function OptForm({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      {children}
    </Container>
  )
}

OptForm.Text = function OptFormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

OptForm.Input = function OptFromInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>
}

OptForm.Button = function OptFormButton({ children, ...restProps }) {
  return (
    <Button {...restProps}>
      {children} <img src='/images/icons/chevron-right.png' alt='Get Started' />
    </Button>
  )
}

OptForm.Break = function OptFormBreak({ children, ...restProps }) {
  return <Break {...restProps} />
}