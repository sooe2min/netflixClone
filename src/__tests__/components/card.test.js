import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Card, Player } from '../../components'

const category = 'series'
const slideRows = [
  {
    title: 'Documentaries',
    data: [
      {
        id: '07adcf19-517f-41b4-8f8b-04fee694bce1',
        title: 'Tiger King',
        description: 'Tiger King description',
        genre: 'documentaries',
        maturity: '18',
        slug: 'tiger-king',
        docId: 'IcRxJBanzE7nmIp2GMGb'
      }
    ]
  },
  {
    title: 'Children',
    data: [
      {
        id: 'a938b5a6-fe25-4a06-8050-353bc7146ccd',
        title: 'Hotel Transylvania',
        description: 'Hotel Transylvania description',
        genre: 'children',
        maturity: '0',
        slug: 'hotel-transylvania',
        docId: '4JDgdf5vE0K3YrW9ZnbP'
      }
    ]
  }
]

describe('<Card />', () => {
  it('renders the <Card /> with populated data', () => {
    const { container, getByText } = render(
      <Card.Group>
        {slideRows.map(slideItem => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map(item => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Image src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`} />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src='/videos/bunny.mp4' />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    )

    expect(getByText('Documentaries')).toBeTruthy()
    expect(getByText('Tiger King')).toBeTruthy()
    expect(getByText('Tiger King description')).toBeTruthy()

    expect(getByText('Children')).toBeTruthy()
    expect(getByText('Hotel Transylvania')).toBeTruthy()
    expect(getByText('Hotel Transylvania description')).toBeTruthy()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders the <Card /> with and clicks the card feature', () => {
    const { container, queryByText, getByTestId, getByAltText } = render(
      <Card.Group>
        {slideRows.map(slideItem => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map(item => (
                <Card.Item key={item.docId} item={item} data-testid={`${item.slug}-testid`}>
                  <Card.Image src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`} />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src='/videos/bunny.mp4' />
              </Player>
            </Card.Feature>
          </Card>
        ))
        }
      </Card.Group >
    )

    expect(queryByText('18')).toBeFalsy()
    fireEvent.click(getByTestId('tiger-king-testid'))
    expect(queryByText('18')).toBeTruthy()

    fireEvent.click(getByAltText('Close'))
    expect(queryByText('18')).toBeFalsy()

    expect(queryByText('PG')).toBeFalsy()
    fireEvent.click(getByTestId('hotel-transylvania-testid'))
    expect(queryByText('PG')).toBeTruthy()

    fireEvent.click(getByAltText('Close'))
    expect(queryByText('PG')).toBeFalsy()

    expect(container.firstChild).toMatchSnapshot()
  })
})
