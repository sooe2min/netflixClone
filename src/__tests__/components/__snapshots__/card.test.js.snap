// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders the <Card /> with and clicks the card feature 1`] = `
<div
  class="card__Group-s87lnr-2 eqsXHQ"
>
  <div
    class="card__Container-s87lnr-1 bXGGHr"
  >
    <p
      class="card__Title-s87lnr-0 jKOXfe"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-s87lnr-3 eIZgGz"
    >
      <div
        class="card__Item-s87lnr-7 hDzbZC"
        data-testid="tiger-king-testid"
      >
        <img
          class="card__Image-s87lnr-8 cWYycC"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-s87lnr-4 cKCXxn"
        >
          <p
            class="card__SubTitle-s87lnr-5 eifFOc"
          >
            Tiger King
          </p>
          <p
            class="card__Text-s87lnr-6 cqvlwl"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-s87lnr-1 bXGGHr"
  >
    <p
      class="card__Title-s87lnr-0 jKOXfe"
    >
      Children
    </p>
    <div
      class="card__Entities-s87lnr-3 eIZgGz"
    >
      <div
        class="card__Item-s87lnr-7 hDzbZC"
        data-testid="hotel-transylvania-testid"
      >
        <img
          class="card__Image-s87lnr-8 cWYycC"
          src="/images/series/children/hotel-transylvania/small.jpg"
        />
        <div
          class="card__Meta-s87lnr-4 cKCXxn"
        >
          <p
            class="card__SubTitle-s87lnr-5 eifFOc"
          >
            Hotel Transylvania
          </p>
          <p
            class="card__Text-s87lnr-6 cqvlwl"
          >
            Hotel Transylvania description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card /> renders the <Card /> with populated data 1`] = `
<div
  class="card__Group-s87lnr-2 eqsXHQ"
>
  <div
    class="card__Container-s87lnr-1 bXGGHr"
  >
    <p
      class="card__Title-s87lnr-0 jKOXfe"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-s87lnr-3 eIZgGz"
    >
      <div
        class="card__Item-s87lnr-7 hDzbZC"
      >
        <img
          class="card__Image-s87lnr-8 cWYycC"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-s87lnr-4 cKCXxn"
        >
          <p
            class="card__SubTitle-s87lnr-5 eifFOc"
          >
            Tiger King
          </p>
          <p
            class="card__Text-s87lnr-6 cqvlwl"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-s87lnr-1 bXGGHr"
  >
    <p
      class="card__Title-s87lnr-0 jKOXfe"
    >
      Children
    </p>
    <div
      class="card__Entities-s87lnr-3 eIZgGz"
    >
      <div
        class="card__Item-s87lnr-7 hDzbZC"
      >
        <img
          class="card__Image-s87lnr-8 cWYycC"
          src="/images/series/children/hotel-transylvania/small.jpg"
        />
        <div
          class="card__Meta-s87lnr-4 cKCXxn"
        >
          <p
            class="card__SubTitle-s87lnr-5 eifFOc"
          >
            Hotel Transylvania
          </p>
          <p
            class="card__Text-s87lnr-6 cqvlwl"
          >
            Hotel Transylvania description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
